### PR TITLE
chore(tooltip): new component

### PR DESCRIPTION
**Pull request by rachelbt**
image

Clean and functional tooltip.
Uses the popup component.
No need to review the popup changes :)
test are still needed :)

### CODE REVIEW

**Review by yinonov — COMMENTED** [review]
A tooltip provides additional information about an interacted target. It's commonly shown on mouse hover or focus. Do not target non-interactive controls as a tooltip's anchor (such as non-focusable / disabled elements). The absence of user interaction will prevent the tooltip from showing up.

Note that touch screen devices do not share the same pointer input events and may fail to show anything.

Consider avoiding the use of tooltip as it may fail to produce the same UX for ALL users. if must, contemplate on practices such as _aria-describedby_ to add alternative experience to users who aren't able to interact with a control in the same manner as sighted users

File: components/tooltip/src/vwc-tooltip-base.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import {
+	html, LitElement, property
+} from 'lit-element';
+import type { TemplateResult } from 'lit-html';
+import type { Placement } from '@floating-ui/core';
+
+export class VWCTooltipBase extends LitElement {
+
+	/**
+	 * @prop content - the content of the tooltip
```
Comment (rinaok) [inline components/tooltip/src/vwc-tooltip-base.ts:10]:
```suggestion
	 * @prop text - the text of the tooltip
```

File: components/tooltip/src/vwc-tooltip-base.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	/**
+	 * @prop corner - the placement of the tooltip
+	 * accepts top, top-start, top-end, bottom, bottom-start, bottom-end, right, right-start, right-end, left, left-start, left-end
+	 * @public
+	 * */
+	@property({ type: String, reflect: true })
+		corner?: Placement;
+
+	/**
+	 * @prop open - indicates whether the tip is open
```
Comment (rinaok) [inline components/tooltip/src/vwc-tooltip-base.ts:26]:
```suggestion
	 * @prop open - indicates whether the tooltip is open
```

File: components/tooltip/src/vwc-tooltip-base.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	 * accepts boolean value
```
Comment (rinaok) [inline components/tooltip/src/vwc-tooltip-base.ts:27]:
```suggestion
	 * accepts boolean value
	  * @public
```

File: components/tooltip/src/vwc-tooltip-base.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	/**
+	 * Opens the popup
+	 * @public
+	 */
+	show(): void {
+		this.open = true;
+	}
+
+	/**
+	 * Closes the popup
```
Comment (rinaok) [inline components/tooltip/src/vwc-tooltip-base.ts:49]:
```suggestion
	 * Closes the tooltip
```

File: components/tooltip/src/vwc-tooltip-base.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	/**
+	 * @prop anchor - ID reference to element in the popup’s owner document.
+	 * accepts string
+	 * @public
+	 * */
+	@property({ type: String })
+		anchor = '';
+
+	/**
+	 * Opens the popup
```
Comment (rinaok) [inline components/tooltip/src/vwc-tooltip-base.ts:41]:
```suggestion
	 * Opens the tooltip
```

**Comment by rinaok** [comment]
Screen Shot 2022-02-01 at 15 25 12

**Comment by rinaok** [comment]
Except for the UI test, there are no other tests.

**Review by rinaok — CHANGES_REQUESTED** [review]
Except for the UI test, there are no other tests.

**Comment by rachelbt** [comment]
> Screen Shot 2022-02-01 at 15 25 12

there is not such a long word - if there to be longer text it will break into lines:
image

**Comment by rachelbt** [comment]
> A tooltip provides additional information about an interacted target. It's commonly shown on mouse hover or focus. Do not target non-interactive controls as a tooltip's anchor (such as non-focusable / disabled elements). The absence of user interaction will prevent the tooltip from showing up.
> 
> Note that touch screen devices do not share the same pointer input events and may fail to show anything.
> 
> Consider avoiding the use of tooltip as it may fail to produce the same UX for ALL users. if must, contemplate on practices such as _aria-describedby_ to add alternative experience to users who aren't able to interact with a control in the same manner as sighted users

Added to the readme - separated some of the text.

**Comment by rinaok** [comment]
> Except for the UI test, there are no other tests.

@rachelbt I still don't see a `test` folder inside tooltip folder.

**Review by rinaok — CHANGES_REQUESTED** [review]
@rachelbt I still don't see a test folder inside tooltip folder.

**Review by rinaok — DISMISSED** [review]

File: components/tooltip/stories/tooltip.stories.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	</style>
+	<div class="tooltip-wrapper" >
+			<vwc-text style="display: flex; align-items: center;" font-face="body-1" tight>text with tooltip - press the question mark
+			<vwc-icon-button style="margin: 0 0.125rem;" icon="info-line"  shape="circled" aria-describedby="tooltip" id="button" dense @click=${onClick}></vwc-icon-button>
+		 more text after tooltip</vwc-text>
+		<vwc-tooltip ...=${spread(args)} icon="info-line" style="--tooltip-inline-size:180px;" id="tooltip" anchor="button" text="I'm the tooltip content"></vwc-tooltip>
+	</div>
+`;
+
+export const InText = InsideTExtTemplate.bind({});
```
Comment (yinonov) [inline components/tooltip/stories/tooltip.stories.js:64]:
```suggestion
export const InText = InsideTextTemplate.bind({});
```

File: components/tooltip/stories/tooltip.stories.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	<div class="tooltip-wrapper" >
+   <div style="display: flex; align-items: center;">
+			<vwc-text font-face="title-2" tight><p class="line" style="margin-right: 0.25rem">I'm a title - press the question mark</p></vwc-text>
+			<vwc-icon-button icon="info-line"  shape="circled" aria-describedby="tooltip" id="button" @click=${onClick}></vwc-icon-button>
+		 <div>
+		<vwc-tooltip ...=${spread(args)} icon="info-line"  style="--tooltip-inline-size:180px;" id="tooltip" anchor="button" text="I'm the tooltip content"></vwc-tooltip>
+	</div>
+`;
+
+export const InATitle = AfterATitleTemplate.bind({});
```
Comment (yinonov) [inline components/tooltip/stories/tooltip.stories.js:46]:
```suggestion
export const InATitle = AfterTitleTemplate.bind({});
```

File: components/tooltip/stories/tooltip.stories.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	<div class="tooltip-wrapper" >
+		<vwc-icon-button icon="info-line"  shape="circled" aria-describedby="tooltip" id="button" @click=${onClick}></vwc-icon-button>
+		<vwc-tooltip ...=${spread(args)} icon="info-line"  style="--tooltip-inline-size:180px;" id="tooltip" anchor="button" text="I'm the tooltip content", ></vwc-tooltip>
+	</div>
+	`;
+
+export const Basic = Template.bind({});
+Basic.args = { corner: "top" };
+
+const AfterATitleTemplate = args => html`
```
Comment (yinonov) [inline components/tooltip/stories/tooltip.stories.js:30]:
```suggestion
const AfterTitleTemplate = args => html`
```

**Review by yinonov — CHANGES_REQUESTED** [review]

**Review by yinonov — APPROVED** [review]

**Review by rinaok — APPROVED** [review]

**Pull request merged**